R/generate.R

```diff
@@ -0,0 +1,55 @@
+int_types <- c("integer", "smallint")
+logi_types <- c("boolean")
+num_types <- c("date", "numeric", "real")
+date_types <- c("timestamp without time zone", "timestamp with time zone")
+
+generate_type_check <- function(column_name, data_type, is_nullable) {
+  r_type <- ifelse(data_type %in% int_types, "\"integer\"",
+                   ifelse(data_type %in% logi_types, "\"logical\"",
+                          ifelse(data_type %in% num_types, "\"double\"", "\"character\"")))
+
+  type_check <- paste0("typeof(", column_name, ") == ", r_type)
+  type_check <- ifelse(data_type %in% date_types, paste0("inherits(", column_name, ", \"POSIXct\")"), type_check)
```

Comment:
Generally prefer `sprintf` (or glue) to string concatenation.  you could write this as

```
type_check <- if (data_type %in% date_types) sprintf('inherits(%s, "POSIXct")', column_name) else type_check
```

However, I'm finding the logic here with the `type_check` definition above hard to read.  I think this is really

```suggestion
  if (data_type %in% date_type) {
    type_check <- sprintf('inherits(%s, "POSIXct")', column_name)
  } else {
    type_check <- sprintf("typeof(%s) == %s", column_name, r_type)
  }
```

If you avoided quoting in the block starting on line 7 the format in the else clause would be `'typeof(%s) == "%s"'`